project_report/project_report_latex.py

```diff
@@ -0,0 +1,502 @@
+                        self.doc.append(NoEscape('\n'))
+                        self.doc.append('More detail about the format in ')
+                        self.doc.append(HRef(url=report_text.get('vcf_link'), text='VCF specification'))
+
+        self.doc.append(NewPage())
+
+    def create_formal_statement_section(self, project_name, authorisations):
+        with self.doc.create(Section('Deviations, Additions and Exclusions', numbering=True)):
+            for authorisation in authorisations:
+                if 'NCs' in authorisation and authorisation.get('NCs'):
+                    title = '{project} {version}: {date}'.format(
+                        project=project_name, version=authorisation.get('version'), date=authorisation.get('date')
+                    )
+                    with self.doc.create(Subsection(title, numbering=True)):
+                        self.doc.append(authorisation.get('NCs'))
+        with self.doc.create(Section('Declaration of Compliance', numbering=True)):
+            add_text(self.doc, report_text.get('formal_statement'))
+
+    def create_appendix_tables(self, appendix_tables):
+        with self.doc.create(Section('Appendix I. Per sample metadata', numbering=True)):
+            add_text(self.doc, report_text.get('appendix_description'))
+            self.doc.append(LineBreak())
+
+            with self.doc.create(LatexSection('scriptsize',)) as small_section:
+                self.create_vertical_table(
+                    small_section,
+                    appendix_tables['appendix I']['header'],
+                    self._limit_cell_width(appendix_tables['appendix I']['rows'], {0: 40}),
+                    # Set the column width to fix width for all but first column
+                    column_def='X[l] p{2.5cm} p{2cm} p{2cm} p{1.7cm} p{2cm}',
+                    footer=appendix_tables['appendix I']['footer']
+                )
+        self.doc.append(NewPage())
+
+        with self.doc.create(Section('Appendix II. Per Sample Results', numbering=True)):
+            add_text(self.doc, report_text.get('appendix_description'))
+            self.doc.append(LineBreak())
+
+            with self.doc.create(LatexSection('scriptsize',)) as small_section:
+                self.create_vertical_table(
+                    small_section,
+                    appendix_tables['appendix II']['header'],
+                    self._limit_cell_width(appendix_tables['appendix II']['rows'], {0: 40}),
+                    # Set the column width to fix width for all but first column
+                    # R: is the newly defined column type in the preamble
+                    'X[l] R{2cm} R{2.2cm} R{1.9cm} R{2cm} R{2cm}'
+                )
+
+    def front_page(self, project_name, report_version, authorisations):
+        with self.doc.create(MiniPage(height='5cm', pos='c', align='c')) as logo_wrapper:
+            logo_wrapper.append(HRef(
+                url='https://genomics.ed.ac.uk',
```

Comment:
This could be reference from an external file.

Reply:
added to the `report_text.yaml`